subsys/bluetooth/host/mesh/prov.c

```diff
@@ -1067,20 +1067,20 @@ static void prov_data(const u8_t *data)
 	link.expect = 0;
 
 	/* Store info, since bt_mesh_provision() will end up clearing it */
-	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+#ifdef CONFIG_BT_MESH_GATT_PROXY
```

Comment:
IS_ENABLED() is used extensively throughout the code base, so I'd be surprised if that's the issue here. I suspect it's the intermediate variable `identity_enable` that confuses the linker.

Reply:
Yes, that variable is actually the culprit.
But isn't it better to avoid IS_ENABLED. this just adds more branch statements. By using #ifdef blocks we can optimize the code manually. instead of assuming the compiler will do it for us.